ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -5560,12 +5604,16 @@ public void cacheFileMetadata(
 
   public void dropConstraint(String dbName, String tableName, String constraintName)
     throws HiveException, NoSuchObjectException {
+
     try {
+      Table tbl = getTable(dbName, tableName);
+
```

Comment:
Why is this change needed?

Reply:
Rebased with master and only kept relevant changes.